src/client.js

```diff
@@ -2844,16 +2844,24 @@ function _sendEventHttpRequest(client, event) {
  * @param {string} eventId
  * @param {string} [txnId]  transaction id. One will be made up if not
  *    supplied.
+ * @param {string|module:client.callback} reason Optional. Can be callback for backwards compatibility.
  * @param {module:client.callback} callback Optional.
  * @return {Promise} Resolves: TODO
  * @return {module:http-api.MatrixError} Rejects: with an error response.
  */
-MatrixClient.prototype.redactEvent = function(roomId, eventId, txnId, callback) {
+MatrixClient.prototype.redactEvent = function(roomId, eventId, txnId, reason, callback) {
```

Comment:
this isn't backwards compatible :( 

I was trying to say something along the lines of:

```
function(roomId, eventId, txnId, reason, callbackOrOpts) {
  const opts = typeof(callbackOrOpts) === 'object' ? callbackOrOpts : {};
  const reason = opts['reason'];
  const callback = typeof(callbackOrOpts) === 'function' ? callbackOrOpts : undefined;
  // ... the rest of the function
}
```

Reply:
we don't need to worry about supplying a callback as a 5th argument when `opts` is used because we're slowly trying to discourage people from using the callback.

Reply:
Got it, so object and deprecating callbacks it shall be - I'll adjust it when I get around to it again. :-)